Mathlib/Analysis/Normed/Group/Quotient.lean

```diff
@@ -0,0 +1,532 @@
+
+instance Submodule.Quotient.normedAddCommGroup [hS : IsClosed (S : Set M)] :
+    NormedAddCommGroup (M ⧸ S) :=
+  @AddSubgroup.normedAddCommGroupQuotient _ _ S.toAddSubgroup hS
+#align submodule.quotient.normed_add_comm_group Submodule.Quotient.normedAddCommGroup
+
+instance Submodule.Quotient.completeSpace [CompleteSpace M] : CompleteSpace (M ⧸ S) :=
+  QuotientAddGroup.completeSpace M S.toAddSubgroup
+#align submodule.quotient.complete_space Submodule.Quotient.completeSpace
+
+/-- For any `x : M ⧸ S` and any `0 < ε`, there is `m : M` such that `Submodule.Quotient.mk m = x`
+and `‖m‖ < ‖x‖ + ε`. -/
+nonrec theorem Submodule.Quotient.norm_mk_lt {S : Submodule R M} (x : M ⧸ S) {ε : ℝ} (hε : 0 < ε) :
+    ∃ m : M, Submodule.Quotient.mk m = x ∧ ‖m‖ < ‖x‖ + ε :=
+  norm_mk_lt x hε
+#align submodule.quotient.norm_mk_lt Submodule.Quotient.norm_mk_lt
+
+theorem Submodule.Quotient.norm_mk_le (m : M) : ‖(Submodule.Quotient.mk m : M ⧸ S)‖ ≤ ‖m‖ :=
+  quotient_norm_mk_le S.toAddSubgroup m
+#align submodule.quotient.norm_mk_le Submodule.Quotient.norm_mk_le
+
+instance Submodule.Quotient.normedSpace (𝕜 : Type _) [NormedField 𝕜] [NormedSpace 𝕜 M] [SMul 𝕜 R]
+    [IsScalarTower 𝕜 R M] : NormedSpace 𝕜 (M ⧸ S) :=
+  { Submodule.Quotient.module' S with
+    norm_smul_le := fun k x =>
+      -- porting note: this is `QuotientAddGroup.norm_lift_apply_le` for `f : M → M ⧸ S` given by
+      -- `x ↦ mk (k • x)`; todo: add scalar multiplication as `NormedAddGroupHom`, use it here
+      le_of_forall_pos_le_add fun ε hε => by
+        have := (nhds_basis_ball.tendsto_iff nhds_basis_ball).mp
+          ((@Real.uniformContinuous_const_mul ‖k‖).continuous.tendsto ‖x‖) ε hε
+        simp only [mem_ball, exists_prop, dist, abs_sub_lt_iff] at this
+        rcases this with ⟨δ, hδ, h⟩
+        obtain ⟨a, rfl, ha⟩ := Submodule.Quotient.norm_mk_lt x hδ
+        specialize h ‖a‖ ⟨by linarith, by linarith [Submodule.Quotient.norm_mk_le S a]⟩
+        calc
+          _ ≤ ‖k‖ * ‖a‖ := (quotient_norm_mk_le S.toAddSubgroup (k • a)).trans_eq (norm_smul k a)
+          _ ≤ _ := (sub_lt_iff_lt_add'.mp h.1).le }
+#align submodule.quotient.normed_space Submodule.Quotient.normedSpace
+
+end Submodule
+
+section Ideal
+
+variable {R : Type _} [SeminormedCommRing R] (I : Ideal R)
+
+set_option synthInstance.etaExperiment true in -- Porting note: gets around lean4#2074
+nonrec theorem Ideal.Quotient.norm_mk_lt {I : Ideal R} (x : R ⧸ I) {ε : ℝ} (hε : 0 < ε) :
+    ∃ r : R, Ideal.Quotient.mk I r = x ∧ ‖r‖ < ‖x‖ + ε :=
+  norm_mk_lt x hε
+#align ideal.quotient.norm_mk_lt Ideal.Quotient.norm_mk_lt
+
+set_option synthInstance.etaExperiment true in -- Porting note: gets around lean4#2074
+theorem Ideal.Quotient.norm_mk_le (r : R) : ‖Ideal.Quotient.mk I r‖ ≤ ‖r‖ :=
+  quotient_norm_mk_le I.toAddSubgroup r
+#align ideal.quotient.norm_mk_le Ideal.Quotient.norm_mk_le
+
+set_option synthInstance.etaExperiment true in -- Porting note: gets around lean4#2074
```

Comment:
```suggestion
set_option synthInstance.etaExperiment true in -- Porting note: gets around lean4#2074
set_option maxHeartbeats 400000 in
```
How bad would increasing the heartbeat limit to `400000` for the `semiNormedCommRing` instance and to `800000`  for the `normedAlgebra` instance be? It would unblock this PR, and could be reverted when the reenableeta change lands.